apps/emqx_mt/src/emqx_mt_state.erl

```diff
@@ -321,6 +321,10 @@ with_ccache(Ns) ->
     case lookup_counter_from_cache(Ns) of
         false ->
             update_ccache(Ns);
+        Cnt when Cnt < 1_000 ->
```

Comment:
Should check the total size of the cache. Otherwise this select count can be very expensive when there are a lot of total clients but only a few for a specific ns

Reply:
Actually, it would be nice to have some tests to see how `do_count_clients/1` behaves when there are large amount of namespaces.

Reply:
After our discussion, we decided on changing the table types to `ordered_set`, so that `ets:select` with partially bound key patterns will be efficient.

https://github.com/emqx/emqx/pull/15841/commits/099d74b855f3b58dbe249a8696a127b01d31efbc